arbor/fvm_layout.cpp

```diff
@@ -858,14 +858,17 @@ fvm_mechanism_data fvm_build_mechanism_data(const cable_cell_global_properties&
         const std::string& name = entry.first;
         mechanism_info info = catalogue[name];
 
-        std::vector<double> param_dflt;
         fvm_mechanism_config config;
-        config.kind = arb_mechanism_kind_density;
+        if (info.kind != arb_mechanism_kind_density) {
+            throw cable_cell_error("Expected density mechanism, got "+ name);
```

Comment:
For all these errors it might help the user if the actual `kind` is included in the message, eg
```
Expected density mechanism, got <NAME> which is of kind <KIND>.
```
It's nothing that can't be checked with a Python REPL is 30s, but I think it would be appreciated.